sigstore/sign.py

```diff
@@ -192,6 +231,75 @@ def sign(
         )
 
 
+class SigningContext:
+    """
+    Keep a context between signing operations.
+    """
+
+    def __init__(
+        self,
+        *,
+        fulcio: FulcioClient,
+        rekor: RekorClient,
+    ):
+        """
+        Create a new `SigningContext`.
+
+        `fulcio` is a `FulcioClient` capable of connecting to a Fulcio instance
+        and returning signing certificates.
+
+        `rekor` is a `RekorClient` capable of connecting to a Rekor instance
+        and creating transparency log entries.
+        """
+        self._fulcio = fulcio
+        self._rekor = rekor
+
+    @classmethod
+    def production(cls) -> SigningContext:
+        """
+        Return a `SigningContext` instance configured against Sigstore's production-level services.
+        """
+        updater = TrustUpdater.production()
+        rekor = RekorClient.production(updater)
+        return cls(
+            fulcio=FulcioClient.production(),
+            rekor=rekor,
+        )
+
+    @classmethod
+    def staging(cls) -> SigningContext:
+        """
+        Return a `SignerContext` instance configured against Sigstore's staging-level services.
+        """
+        updater = TrustUpdater.staging()
+        rekor = RekorClient.staging(updater)
+        return cls(
+            fulcio=FulcioClient.staging(),
+            rekor=rekor,
+        )
+
+    @contextmanager
+    def with_signer(
+        self, identity_token: str, cache: bool = True
+    ) -> Generator[Signer, None, None]:
```

Comment:
Nitpick: `mypy` lets us get away with this 🙂 

```suggestion
    ) -> Iterator[Signer]:
```

Reply:
Nice! Will replace it